src/python-common/ceph/deployment/service_spec.py

```diff
@@ -778,3 +780,121 @@ def __init__(self,
 
 
 yaml.add_representer(AlertManagerSpec, ServiceSpec.yaml_representer)
+
+
+class CustomContainerSpec(ServiceSpec):
+    props: List[str] = [
+        'image', 'entrypoint', 'uid', 'gid', 'args', 'volume_mounts',
+        'container_args', 'envs', 'privileged', 'ptrace', 'bind_mounts',
+        'ports', 'dirs', 'files'
+    ]
+
+    def __init__(self,
+                 service_type: str = 'container',
+                 service_id: str = None,
+                 placement: Optional[PlacementSpec] = None,
+                 unmanaged: bool = False,
+                 preview_only: bool = False,
+                 image: str = None,
+                 entrypoint: Optional[str] = None,
+                 uid: Optional[int] = None,
+                 gid: Optional[int] = None,
+                 args: Optional[List[str]] = [],
+                 volume_mounts: Optional[Dict[str, str]] = {},
+                 container_args: Optional[List[str]] = [],
+                 envs: Optional[List[str]] = [],
+                 privileged: Optional[bool] = False,
+                 bind_mounts: Optional[List[List[str]]] = None,
+                 ptrace: Optional[bool] = False,
+                 ports: Optional[List[int]] = [],
+                 dirs: Optional[List[str]] = [],
+                 files: Optional[Dict[str, Any]] = {}
+                 ):
+        assert service_type == 'container'
+        assert service_id is not None
+        assert image is not None
+        super(CustomContainerSpec, self).__init__(
+            service_type, service_id,
+            placement=placement, unmanaged=unmanaged,
+            preview_only=preview_only)
+
+        # Options:
+        # --------
+        # - image: The name of the Docker Hub image.
+        # - entrypoint: Overwrite the default ENTRYPOINT of the image.
+        # - args: Additional cephadm command line arguments.
```

Comment:
I'd like to avoid `args`. The call to cephadm should hopefully be transparent to the spec.